File: Leet_Code-1041/Leet_Code-1041.java
/*
 * Question 1041
 */

class Solution {
    public boolean isRobotBounded(String instructions) {
        int[] position = {0,0};
        int[] p = {0,0};
        Map<Integer, int[]> directions = new HashMap<>();
        directions.put(0, new int[]{0,1});
        directions.put(1, new int[]{-1,0});
        directions.put(2, new int[]{0,-1});
        directions.put(3, new int[]{1,0});
        int i = 0;
        for (int n = 0; n <5; n++){
            for(char l : instructions.toCharArray()){
                if (l == 'L' && i<3){
                    i+=1;
                } else if (l == 'L' && i==3){
                    i=0;
                } else if (l == 'R' && i>0){
                    i-=1;
                } else if (l == 'R' && i==0){
                    i=3;
                } else{
                    position[0] += directions.get(i)[0];
                    position[1] += directions.get(i)[1];
                }
            }
            if (Arrays.equals(position,p)){
                return true;
            }
        }
        return false;
    }
}

File: Leet_Code-1041/Leet_Code-1041.py
'''
Question 1041
'''

class Solution:
    def isRobotBounded(self, instructions: str) -> bool:
        position = [0,0]
        directions = {0: [0,1], 1:[-1,0], 2:[0,-1], 3:[1,0]}
        i = 0
        for n in range(5):
            for l in instructions:
                if l == 'L' and i < 3:
                    i+=1
                elif l == 'L' and i == 3:
                    i = 0
                elif l == 'R' and i>0:
                    i -= 1
                elif l == 'R' and i == 0:
                    i = 3
                else:
                    position[0] += directions[i][0]
                    position[1] += directions[i][1]
            if position == [0,0]:
                return True
        return False